### DISCUSSION TITLE
Use Legend in Node JS/TS application

### DISCUSSION THREAD

**Discussion by idaromnet** [LegendApp/legend-state#242] [General]
Hello!

Apparently, it is not possible to use Legend inside Node JS ( I only use it for listen to changes). I basically get the following errors.

`node_modules/@legendapp/state/src/observableInterfaces.d.ts:1:41 - error TS2307: Cannot find module '@react-native-async-storage/async-storage' or its corresponding type declarations.

1 import type { AsyncStorageStatic } from '@react-native-async-storage/async-storage';
                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@legendapp/state/src/observableInterfaces.d.ts:3:42 - error TS2307: Cannot find module 'firebase/database' or its corresponding type declarations.

3 import { DatabaseReference, Query } from 'firebase/database';`

**Comment by jmeistrich** [comment]
This should be fixed in 2.1.6. But please let me know if it's still not working right!

**Reply by idaromnet** [reply]
Hello !
I installed the new version ("@legendapp/state": "^2.1.6") and got the following errors now :

node_modules/@legendapp/state/src/observableInterfaces.d.ts:2:33 - error TS2307: Cannot find module 'react-native-mmkv' or its corresponding type declarations.

2 type MMKVConfiguration = import('react-native-mmkv').MMKVConfiguration;
                                  ~~~~~~~~~~~~~~~~~~~

node_modules/@legendapp/state/src/observableInterfaces.d.ts:3:34 - error TS2307: Cannot find module '@react-native-async-storage/async-storage' or its corresponding type declarations.

3 type AsyncStorageStatic = import('@react-native-async-storage/async-storage').AsyncStorageStatic;
                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@legendapp/state/src/observableInterfaces.d.ts:4:33 - error TS2307: Cannot find module 'firebase/database' or its corresponding type declarations.

4 type DatabaseReference = import('firebase/database').DatabaseReference;
                                  ~~~~~~~~~~~~~~~~~~~

node_modules/@legendapp/state/src/observableInterfaces.d.ts:5:21 - error TS2307: Cannot find module 'firebase/database' or its corresponding type declarations.

5 type Query = import('firebase/database').Query;
